src/shared/libs/iota/quorum.js

```diff
@@ -0,0 +1,415 @@
+import map from 'lodash/map';
+import filter from 'lodash/filter';
+import includes from 'lodash/includes';
+import isEmpty from 'lodash/isEmpty';
+import isUndefined from 'lodash/isUndefined';
+import transform from 'lodash/transform';
+import size from 'lodash/size';
+import sampleSize from 'lodash/sampleSize';
+import union from 'lodash/union';
+import uniq from 'lodash/uniq';
+import IOTA from 'iota.lib.js';
+import { isNodeSynced } from './extendedApi';
+import {
+    QUORUM_THRESHOLD,
+    QUORUM_SIZE,
+    QUORUM_NODES_SYNC_CHECKS_INTERVAL,
+    DEFAULT_BALANCES_THRESHOLD,
+} from '../../config';
+import { EMPTY_HASH_TRYTES } from './utils';
+import { findMostFrequent } from '../utils';
+import Errors from '../errors';
+
+/**
+ * Resolves only if size of synced nodes is greater than or equal to quorum size
+ *
+ * @method rejectIfNotEnoughSyncedNodes
+ *
+ * @param {array} nodes
+ * @param {number} quorumSize
+ * @returns {Promise}
+ */
+const rejectIfNotEnoughSyncedNodes = (nodes, quorumSize = QUORUM_SIZE) => {
+    if (size(nodes) < quorumSize) {
+        return Promise.reject(new Error(Errors.NOT_ENOUGH_SYNCED_NODES));
+    }
+
+    return Promise.resolve();
+};
+
+/**
+ *   Determines quorum result for supported methods.
+ *
+ *   @method determineQuorumResult
+ *   @param {array} validResults
+ *   @param {number} faultyResultCount
+ *
+ *   @returns {function}
+ **/
+const determineQuorumResult = (validResults, quorumSize = QUORUM_SIZE) => {
+    const { frequency, mostFrequent } = findMostFrequent(validResults);
+
+    return (method, threshold) => {
+        // Include faulty result count while determining percentage
+        const percentage = frequency[mostFrequent] / quorumSize * 100;
```

Comment:
even though it evaluates correctly. I think this is clearer:
```suggestion
        const percentage = (frequency[mostFrequent] / quorumSize) * 100;
```